gcp_delete_old_images.py

```diff
@@ -57,12 +60,97 @@ def delete_old_images():
     subprocess.run(delete_cmd, check=True)
 
 
+def delete_old_docker_images_helper(delete_images, base_cmd,
+                                    latest_images_manifests,
+                                    with_tags=False):
+    if not delete_images:
+        return
+
+    delete_images = [f"{image['package']}@{image['version']}" for
+                     image in delete_images]
+
+    print(f"deleting docker images {'with tags' if with_tags else 'without tags'}:\n" +
+          '\n'.join(delete_images))
+
+    for image in delete_images:
+        # '--delete-tags' flag shouldn't be added when deleting images without
+        # tags to catch unseen errors
+        delete_cmd = base_cmd + ['delete', '--quiet'] + \
+            (['--delete-tags', image] if with_tags else [image])
+        subprocess.run(delete_cmd, check=True)
+
+
+def get_manifests(images):
+    latest_images_manifests = set()
+    for image in images:
+        podman_cmd = ['podman', 'manifest', 'inspect',
+                      f'{image["package"]}@{image["version"]}']
+        res = subprocess.run(podman_cmd, capture_output=True,
+                             check=True, text=True)
+        manifest = json.loads(res.stdout)
+
+        if manifest.get('manifests', False):
+            image_manifests = manifest['manifests']
+            for image_manifest in image_manifests:
+                latest_images_manifests.add(image_manifest['digest'])
+
+    return latest_images_manifests
+
+
+def delete_old_docker_images():
+    print("\nDeleting docker images")
+    base_cmd = ['gcloud', 'artifacts', 'docker', 'images']
+    get_images_cmd = base_cmd + ['list',
+                                 '--include-tags', '--format',
+                                 'json(package,version,tags)',
+                                 '--filter', 'createTime < -P2W',
+                                 os.environ['GCP_REPO']]
+    res = subprocess.run(get_images_cmd, capture_output=True,
+                         check=True, text=True)
+    old_images = json.loads(res.stdout)
+
+    if len(old_images) == 0:
+        print("no docker images to delete")
+        sys.exit(0)
```

Comment:
I don't think this should use sys.exit(), given that we do multiple things in the script now. I'd just return.